presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -75,3 +79,39 @@ def __validate_position_over_text(self, analyzer_result, text_len):
                 f"Invalid analyzer result: '{analyzer_result}', "
                 f"original text length is only {text_len}."
             )
+
+    def __extract_anonymizer(self, transformation):
+        anonymizer = transformation.get("anonymizer")()
+        # if the anonymizer is not valid, a InvalidParamException
+        anonymizer.validate(params=transformation)
+        return anonymizer
+
+    def __anonymize_section(
+        self, anonymizer, output_text, analyzer_result, transformation
+    ):
+        text_to_anonymize = output_text[analyzer_result.start : analyzer_result.end]
+        anonymized_text = anonymizer.anonymize(
+            params=transformation, text=text_to_anonymize
+        )
+        return anonymized_text
+
+    def __update_output_with_anonymized_section(
+        self, analyzer_result, last_replacement_point, output_text, anonymized_text
+    ):
+        end_of_text = min(analyzer_result.end, last_replacement_point)
+        return (
+            output_text[: analyzer_result.start]
+            + anonymized_text
+            + output_text[end_of_text:]
+        )
+
+    def __get_transformation_from_analyzer_result(
```

Comment:
nit: If this only calls another method and writes a log, maybe we don't need an extra level here and this could stay in the main `anonymize` function?

Reply:
yes, it is abit over kill :) wanted to make the code as clear as possible.